internal/supported_types/supported_types_interface.go

```diff
@@ -0,0 +1,20 @@
+package supported_types
+
+import (
+	"context"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+)
+
+type supportedInstanceTypes struct {
+	Aws struct {
+		Ec2 []string `yaml:"ec2"`
+	} `yaml:"aws"`
+}
+
+var supported supportedInstanceTypes
+
+type SupportedTypes interface {
+	// FilterUnsupportedTypes remove instances which are not supported
+	FilterInstanceTypes(ctx context.Context, is *[]clients.InstanceType) (*[]clients.InstanceType, error)
+}
```

Comment:
this interface is unused :)

Reply:
This is my fault, I talked so much about interfaces that it left an impression we need one here. Not needed actually, we have a package with one exported function actually :)

We could if we would have three different implementation that would be different, but in this case it is just more simple to have a single YAML with three maps there.

Reply:
I think that the interface is needed only if we want to have one - ListInstanceType function and to have a dispatch pattern to revoke the function for a specific cloud provider.

Reply:
Yeah, definitely get rid of the interface for now, also move this into a single file.